core/blockchain.go

```diff
@@ -194,16 +222,21 @@ func ExecuteBlockEphemerallyForBSC(
 		newBlock = block
 	}
 
+	var bloom types.Bloom
+
+	receiptSha := types.DeriveSha(receipts)
```

Comment:
Is this not the same as `newBlock.ReceiptHash()` ? I am just wondering if it gets calculated twice

Reply:
when `!vmConfig.ReadOnly` above, yes it's calculated twice. Refactored to remove that.